week4/homework/Program.cs

```diff
@@ -0,0 +1,123 @@
+var builder = WebApplication.CreateBuilder(args);
+var app = builder.Build();
+
+app.MapGet("/", () => "Hello this is homework 4 World!");
+/*
+A POST endpoint that calls POST users/add with a record with FirstName, 
+LastName and Age (this simulates creating a user)
+*/
+app.MapPost("/create", async (User userData) =>
+{
+    var httpClient = new HttpClient();
+    var response= await httpClient.PostAsJsonAsync("https://dummyjson.com/users/add", userData)
+        ;
+        System.Console.WriteLine(response);
+        var data = await response.Content.ReadFromJsonAsync<User>();
+        var responseData = new
+        {
+            StatusCode = response.StatusCode,
+            ReasonPhrase = response.ReasonPhrase,
+            data
+
+        };
+        return (responseData);
+});
+/*
+A POST endpoint that that calls Post products/add with a record with Title and 
+Price (this simulates creating a product)
+*/
+app.MapPost("/createProduct", async (Product productData) =>
+{
+    var httpClient = new HttpClient();
+    var response= await httpClient.PostAsJsonAsync("https://dummyjson.com/products/add", productData)
+        ;
+        System.Console.WriteLine(response);
+        var data = await response.Content.ReadFromJsonAsync<Product>();
+        var responseData = new
+        {
+            StatusCode = response.StatusCode,
+            ReasonPhrase = response.ReasonPhrase,
+            data
+
+        };
+        return (responseData);
+});
+/*
+A POST endpoint that takes a lists of ids and retrieves 
+all of the users with those ids from the GET users (Id, FirstName, LastName and Age)*/
+app.MapPost("/retrieveUsers", async (UserId ids) =>
+{
+    var httpClient = new HttpClient();
+        var result = "";
+        foreach (var id in ids.Id)
+        
+        {
+            var response= await httpClient.GetAsync($"https://dummyjson.com/users/{id}");
+            var data = await response.Content.ReadFromJsonAsync<User>();
+             result += data.FirstName +" "+ data.LastName+" "+data.Age +" ";

```

Comment:
You could create a list of users here and return that in your endpoint. It is more common to return a list of items as a response than a string or an anonymous object.